app/src/test/java/com/example/android/architecture/blueprints/todoapp/addedittask/AddEditTaskViewModelTest.kt

```diff
@@ -78,9 +83,9 @@ class AddEditTaskViewModelTest {
     }
 
     @Test
-    fun loadTasks_loading() {
-        // Pause dispatcher so we can verify initial values
-        mainCoroutineRule.pauseDispatcher()
+    fun loadTasks_loading() = runTest {
+        // Override eagerly executing Main dispatcher for just a single test
+        Dispatchers.setMain(StandardTestDispatcher())
```

Comment:
Instead of doing this, why not passing the `StandardTestDispatcher` to the `MainCoroutineRule`?

Reply:
All other tests expect eager execution on the Main dispatcher, so some of them would break if we replaced it for the entire test class (or we'd have to make their code more complex by adding things like `advanceUntilIdle`).

The problem here is that with the old APIs we had eager by default which could be cleanly overridden by pause/resume. Now that functionality has be achieved with using a StandardTestDispatcher instead of an UnconfinedTestDispatcher for the main thread in the specific test.

Some alternatives in cases like this would be:
- Using setMain/resetMain directly in each test method, a bit more boilerplate than going with the rule
- Splitting your tests into multiple test classes, depending on which Main dispatcher you need 🙈